x-pack/elastic-agent/pkg/agent/cmd/setup_config.go

```diff
@@ -104,11 +105,12 @@ func defaultAccessConfig() (setupConfig, error) {
 				// Remove FLEET_SETUP in 8.x
 				// The FLEET_SETUP environment variable boolean is a fallback to the old name. The name was updated to
 				// reflect that its setting up Fleet in Kibana versus setting up Fleet Server.
-				Setup:    envBool("KIBANA_FLEET_SETUP", "FLEET_SETUP"),
-				Host:     envWithDefault("http://kibana:5601", "KIBANA_FLEET_HOST", "KIBANA_HOST"),
-				Username: envWithDefault("elastic", "KIBANA_FLEET_USERNAME", "KIBANA_USERNAME", "ELASTICSEARCH_USERNAME"),
-				Password: envWithDefault("changeme", "KIBANA_FLEET_PASSWORD", "KIBANA_PASSWORD", "ELASTICSEARCH_PASSWORD"),
-				CA:       envWithDefault("", "KIBANA_FLEET_CA", "KIBANA_CA", "ELASTICSEARCH_CA"),
+				Setup:        envBool("KIBANA_FLEET_SETUP", "FLEET_SETUP"),
+				Host:         envWithDefault("http://kibana:5601", "KIBANA_FLEET_HOST", "KIBANA_HOST"),
+				Username:     envWithDefault("elastic", "KIBANA_FLEET_USERNAME", "KIBANA_USERNAME", "ELASTICSEARCH_USERNAME"),
+				Password:     envWithDefault("changeme", "KIBANA_FLEET_PASSWORD", "KIBANA_PASSWORD", "ELASTICSEARCH_PASSWORD"),
+				ServiceToken: envWithDefault("", "KIBANA_FLEET_SERVICE_TOKEN", "FLEET_SERVER_SERVICE_TOKEN"),
```

Comment:
As it is the fleet-server service toke, should we indicate this here by calling it `KIBANA_FLEET_SERVER_SERVICE_TOKEN` ?

Reply:
It doesn't have to be a Fleet Server token, it could be any token that has the correct permissions. It could be a different token then the one provided to Fleet Server.

I think its best to keep the name `KIBANA_FLEET_SERVICE_TOKEN` because `KIBANA_FLEET_{name}` is the same for all other environment variables that control Elastic Agent containers communication to Kibana.

It also matches with the configuration that can be used:

```
kibana:
  fleet:
    service_token: abc123
```

Reply:
@joshdover Do we check on the Fleet side if it is a fleet-server service token or if it has the correct permissions?

Reply:
We'll be checking that is has correct permissions, so it technically doesn't have to be the elastic/fleet-server service token. It just needs to have the "application privilege" that is being granted to the service account by default as defined in this issue: https://github.com/elastic/kibana/issues/112647

So by default, Kibana would accept the elastic/fleet-server service account's token or any credentials for a super user (basic, apikey, etc.)